RFS/src/main/java/org/opensearch/migrations/bulkload/common/FilterScheme.java

```diff
@@ -1,26 +1,23 @@
 package org.opensearch.migrations.bulkload.common;
 
 import java.util.List;
-import java.util.function.BiConsumer;
 import java.util.function.Predicate;
 
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
 public class FilterScheme {
     private FilterScheme() {}
 
-    public static Predicate<SnapshotRepo.Index> filterIndicesByAllowList(
-        List<String> indexAllowlist,
-        BiConsumer<String, Boolean> indexNameAcceptanceObserver
-    ) {
-        return index -> {
+    public static Predicate<String> filterByAllowList(List<String> allowlist) {
```

Comment:
Removed the logging features, IMO much cleaner to have the source of the filter know how/when to log directly.